packages/server/lib/browsers/electron.ts

```diff
@@ -283,21 +295,26 @@ export = {
       this._clearCache(win.webContents),
     ])
 
-    await browserCriClient?.currentlyAttachedTarget?.send('Page.enable')
-
-    await Promise.all([
-      protocolManager?.connectToBrowser(cdpAutomation),
-      videoApi && recordVideo(cdpAutomation, videoApi),
-      this._handleDownloads(win, options.downloadsFolder, automation),
-    ])
+    if (cdpAutomation) {
+      await Promise.all([
+        browserCriClient?.currentlyAttachedTarget?.send('Page.enable'),
+        protocolManager?.connectToBrowser(cdpAutomation),
+        cdpSocketServer?.attachCDPClient(cdpAutomation),
+        videoApi && recordVideo(cdpAutomation, videoApi),
+        this._handleDownloads(win, options.downloadsFolder, automation),
+      ])
+    }
 
     // enabling can only happen once the window has loaded
     await this._enableDebugger()
 
-    await win.loadURL(url)
+    if (cdpAutomation) {
```

Comment:
```suggestion
    // Note that these calls have to happen before we load the page so that we don't miss out on any events that happen quickly
    if (cdpAutomation) {
```